infrastructure-scripts/New-SqlDbServiceAccount.ps1

```diff
@@ -32,126 +23,122 @@ The name of the Keyvault for the Environment
 
 .EXAMPLE
 $New-SqlDBAccountParameters = @{
-	 ServerName = ServerName
-     DataBaseName = DataBaseName
-     SqlUserName = SqlUserName
-	 SqlPassword = SqlPassword
-     SqlServiceAccountName = SqlServiceAccountName
-     SQLResourceGroup = SQLResourceGroup
-     Enviroment = Enviroment
-     KeyVaultName = KeyVaultName
+	 ServerName = $ServerName
+     DataBaseName = $DataBaseName
+     SqlServiceAccountName = $SqlServiceAccountName
+     Environment = $Environment
+     KeyVaultName = $KeyVaultName
 }
 
 .\New-SqlDbServiceAccount.ps1 @New-SqlDBAccountParameters
 
 #>
 
-[CmdletBinding(DefaultParameterSetName = 'None')]
-param
-(
-    [Parameter(Mandatory = $true)]
-    [String] $ServerName,
-    [String] $AzureFirewallName = "AzureWebAppFirewall",
+[CmdletBinding()]
+param (
     [Parameter(Mandatory = $true)]
-    [String]  $DataBaseName,
+    [String]$ServerName,
+    [Parameter(Mandatory = $false)]
+    [String]$AzureFirewallName = "AzureWebAppFirewall",
     [Parameter(Mandatory = $true)]
-    [String]$SqlUserName,
+    [String]$DataBaseName,
     [Parameter(Mandatory = $true)]
-    [String]  $SqlPassword,
+    [String]$SqlServiceAccountName,
     [Parameter(Mandatory = $true)]
-    [String]  $SqlServiceAccountName,
+    [String]$Environment,
     [Parameter(Mandatory = $true)]
-    [String]  $SQLResourceGroup,
-    [Parameter(Mandatory = $true)]
-    [String]  $Enviroment,
-    [Parameter(Mandatory = $true)]
-    [String] $KeyVaultName
+    [String]$KeyVaultName
 )
 
 $ErrorActionPreference = 'Stop'
 
-
-function New-AzureSQLServerFirewallRule {
-    $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
-    New-AzureRmSqlServerFirewallRule -StartIPAddress $AgentIp -EndIPAddress $AgentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
-}
-function Update-AzureSQLServerFirewallRule {
-    $AgentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
-    Set-AzureRmSqlServerFirewallRule -StartIPAddress $AgentIp -EndIPAddress $AgentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
-}
-
 function Get-RandomPassword {
     $Password = ([char[]]([char]33..[char]95) + ([char[]]([char]97..[char]126)) + 0..9 | Sort-Object { Get-Random })[0..33] -join ''
     return $Password
 }
 
-function Remove-AzureSQLServerFirewallRule {
-    if ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue)) {
-        Remove-AzureRmSqlServerFirewallRule -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+try {
+	$AgentIP = (Invoke-WebRequest -Uri http://checkip.dyndns.com -UseBasicParsing).Content -Replace "[^\d\.]"
```

Comment:
Have previously used the below to find outbound IP, looks a little neater. Same outcome either way though.
```suggestion
	$AgentIP = (Invoke-WebRequest ifconfig.me/ip -UseBasicParsing).Content.Trim()
```